frontend/src/components/TranslationPageComponent/TranslationPageComponent.js

```diff
@@ -22,55 +30,117 @@ export class TranslationPageComponent extends Component {
     return this.#container;
   }
 
-  // Method to set the list of tasks to display
-  // setTasks(tasks) {
-  //   this.#tasks = tasks;
-  //   this.#renderTasks();
-  // }
-
   // Creates the container element for the component
   #createContainer() {
-    this.#container = document.createElement('div');
-    this.#container.classList.add('translation-view');
+    this.#container = document.createElement("div");
+    this.#container.classList.add("translation-view");
   }
 
   // Sets up the basic HTML structure of the component
   #setupContainerContent() {
     this.#container.innerHTML = `
-      <h2>Translation Page</h2>
+      <div class="container">
+    <h1>TRANSLATOR</h1>
+    <form>
+      <div class="translationSection">
+        <div class="inputSide">
+          <label for="input_lang">Input language</label>
+          <select id="input_lang">
+            <option value="en">English</option>
+            <option value="es">Spanish</option>
+          </select>
+          <label for="input">Input</label>
+          <textarea type="text" id="input"></textarea>
+        </div>
+        <div class="arrow">></div>
+        <br />
+        <div class="outputSide">
+          <label for="output_lang">Output language</label>
+          <select id="output_lang">
+            <option value="en">English</option>
+            <option value="es">Spanish</option>
+          </select>
+          <label for="output">Translated</label>
+          <textarea type="text" id="output" readonly="readonly"></textarea>
+        </div>
+      </div>
+      <div class="translateButton">
+        <input
+          type="button"
+          class="submit-btn"
+          id="translate"
+          value="TRANSLATE"
+        />
+        <input
+          type="button"
+          class="flashcard-btn"
+          id="to_flashcard"
+          value="CREATE FLASHCARD"
+        />
+      </div>
+    </form>
+  </div>
+    `; // TODO add inputs here
+  }
 
-    `;// TODO add inputs here
+  translate() {
+    const input = this.#inputElem.value;
+    const inputLang = this.#inputLangElem.value;
+    const outputLang = this.#outputLangElem.value;
+
+    EventHub.getInstance().publish(Events.Translate, {
```

Comment:
You can define the EventHub as a private variable to follow the convention of other components
`#hub = null`

and then put in the constructor
`this.#hub = EventHub.getInstance()`

and reference the private variable #hub as needed in your functions

Reply:
done